src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy

```diff
@@ -0,0 +1,251 @@
+        }
+
+        and: "Default bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response shouldn't contain seatNonBid and should contain seatbid"
+        assert response.ext.seatnonbid.nonBid.size() == 0
+        assert response.seatbid
+    }
+
+    def "PBS should populate seatNonBid when rejected due to timeout"() {
+        given: "PBS config with min time-out"
+        def pbsService = pbsServiceFactory.getService(
+                ["auction.biddertmax.max": "1",
+                 "auction.biddertmax.min": "1"])
+
+        and: "Default bid request with max timeout"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.returnAllBidStatus = true
+            tmax = 1
+        }
+
+        and: "Default bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response should contain seatNonBid and contain errors"
+        def seatNonBids = response.ext.seatnonbid
+        assert seatNonBids.size() == 1
+
+        def seatNonBid = seatNonBids[0]
+        assert seatNonBid.seat == GENERIC.value
+        assert seatNonBid.nonBid[0].impId == bidRequest.imp[0].id
+        assert seatNonBid.nonBid[0].statusCode == TIMED_OUT
+    }
+
+    def "PBS should populate seatNonBid when filter-imp-media-type = true, imp doesn't contain supported media type"() {
+        given: "Pbs config"
+        def pbsService = pbsServiceFactory.getService(
+                ["auction.filter-imp-media-type.enabled"      : "true",
+                 "adapters.generic.meta-info.site-media-types": "video"])
+
+        and: "Default basic BidRequest with banner"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.returnAllBidStatus = true
+        }
+
+        when: "PBS processes auction request"
+        def response = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response should contains seatNonBid"
+        def seatNonBids = response.ext.seatnonbid
+        assert seatNonBids.size() == 1
+
+        def seatNonBid = seatNonBids[0]
+        assert seatNonBid.seat == GENERIC.value
+        assert seatNonBid.nonBid[0].impId == bidRequest.imp[0].id
+        assert seatNonBid.nonBid[0].statusCode == REJECTED_BY_MEDIA_TYPE
+
+        and: "seatbid should be empty"
+        assert response.seatbid.isEmpty()
+    }
+
+    def "PBS should populate seatNonBid, seatBid when returnAllBidStatus=true and requested bidder didn't bid and did bid"() {
```

Comment:
Test name is not clear. Last part of it to be more exact. Please rephrase this part: "and requested bidder didn't bid and did bid". Because of the wording I can't understand what you are trying to test. There is a possibility that this case is not valid at all.

Reply:
You didn't really make it better :). I still can't understand what exactly test should check based on name.